jina/optimize/__init__.py

```diff
@@ -0,0 +1,230 @@
+import os
+from itertools import tee
+import json
+from pathlib import Path
+import shutil
+from ruamel.yaml import YAML
+
+from jina.flow import Flow
+from jina.helper import colored
+from jina.logging import default_logger as logger
+from .parameters import IntegerParameter, load_optimization_parameters
+
+
+class FlowRunner:
+    def __init__(
+        self,
+        index_document_generator=None,
+        query_document_generator=None,
+        index_batch_size=None,
+        query_batch_size=None,
```

Comment:
Perhaps we can abstract a flow run with kind of a `FlowCompositions`  which we define as `namedtuple('FlowComposition', ['yaml_file', 'document_generator', 'batch_size', 'callback', 'target_endpoint', 'overwrite_workspace'])`, where `target_endpoint` defines, whether we call the `search` or `index` function of the flow. Then you could define your flow runs in a consecutive order and let the FlowRunner just run whichever list of flows you hand over.

Reply:
Ah without the `yaml_file` most probably.

Reply:
Yes we can think on this